.github/copilot-instructions.md

```diff
@@ -0,0 +1,92 @@
+# UGBio Utils - AI Agent Instructions
+
+## Architecture Overview
+
+This is a bioinformatics pipeline toolkit organized as a UV workspace with modular components. Each module in `src/` is a separate Python package with its own Docker container for cloud deployment.
+
+### Core Components
+
+- **`ugbio_core`**: Foundation package providing VCF processing (`vcfbed/`), statistics utilities, and shared infrastructure. All other modules depend on this.
+- **Pipelines**: comparison (Truvari SV analysis), featuremap (VCF→Parquet conversion), filtering, CNV calling, methylation analysis
+- **Cloud Integration**: AWS HealthOmics workflow management, Cromwell cost comparison, distributed processing
+
+### Module Structure Pattern
+```
+src/<module>/
+├── Dockerfile                    # Multi-stage build from ugbio_base
+├── pyproject.toml               # UV workspace member config
+├── ugbio_<module>/              # Python source code
+├── tests/                       # pytest test suite
+└── README.<module>.md           # Module documentation
+```
+
+## Development Workflows
+
+### Environment Setup
+```bash
+# Use UV for all dependency management
+uv sync --all-extras --all-packages  # Full workspace
+uv sync --package ugbio-<module>     # Single module
+
+# Dev containers preferred for testing (includes bioinformatics tools)
+# F1 → "Dev Containers: Open Folder in Container"
+# Choose specific module container (CNV/comparison/etc.)
+```
+
+### Testing Strategy
+```bash
+# In dev container or with proper Python env:
+uv run pytest src/<module>/tests/   # Module tests
+uv run pytest src/core/tests/       # Core functionality
+
+# Docker-based testing:
+docker run --rm -v .:/workdir <image> run_tests /workdir/src/<module>
+```
+
+### Build Process
+- **Local Development**: UV handles Python dependencies and virtual environments
+- **Production**: Multi-stage Docker builds using `ugbio_base` image containing bcftools, samtools, GATK
+- **Dependencies**: Modules declare workspace dependencies via `[tool.uv.sources]`
+
+## Project-Specific Conventions
+
+### Pipeline Architecture
+- **SimplePipeline**: Core execution framework for shell commands
+- **Truvari Integration**: SV comparison using `--passonly` flags (conditional via `ignore_filter`)
+- **Parallel Processing**: Region-based chunking for large genomics files (300Mbp default)
+
+### Testing Requirements
+```python
+# Mock external tools (bcftools, truvari, etc.)
+@patch("subprocess.run")
+@patch("ugbio_core.vcfbed.vcftools.get_vcf_df")
+
+```
+
+### Docker Integration Points
+- **Base Image**: `ugbio_base` provides bcftools, samtools, tabix, common tools
+- **AWS ECR**: `337532070941.dkr.ecr.us-east-1.amazonaws.com/ugbio_<module>`
```

Comment:
is there a need to add google's image repository ?

Reply:
Removed the section